File: data_types.py
"""
This file defines the data types used by the rest of the program.
"""

class MathProblem:
  """Represents a math problem (question and right and wrong answers)"""
  question = None
  right_answer = None
  wrong_answer_1 = None
  wrong_answer_2 = None
  wrong_answer_3 = None

  def __init__(self, question, right_answer, wrong_answer_1, wrong_answer_2, wrong_answer_3):
    self.question = question
    self.right_answer = right_answer
    self.wrong_answer_1 = wrong_answer_1
    self.wrong_answer_2 = wrong_answer_2
    self.wrong_answer_3 = wrong_answer_3

class Equation:
  """Represents an equation with at least one variable"""
  expression = None
  answer = None
  missing_term = None
  
  def __init__(self, expression, answer, missing_term):
    self.expression = expression
    self.answer = answer
    self.missing_term = missing_term

File: dataTypes.js
// This file defines the data types used by the rest of the program.

class MathProblems {
    //Represents a math problem (question and right and wrong answers)
    constructor(question, rightAnswer, wrongAnswer1, wrongAnswer2, wrongAnswer3) {
        this.question = question;
        this.rightAnswer = rightAnswer;
        this.wrongAnswer1 = wrongAnswer1;
        this.wrongAnswer2 = wrongAnswer2;
        this.wrongAnswer3 = wrongAnswer3;
    }

};


class Equation {
    //Represents an equation with at least one variable
    constructor(expression, answer, missingTerm) {
        this.expression = expression;
        this.answer = answer;
        this.missingTerm = missingTerm;
    }

};